integreat_chat/chatanswers/services/search.py

```diff
@@ -74,4 +78,40 @@ def search_documents(
             consistency_level="Strong",
             output_fields=(["source", "text"] if include_text else ["source"])
         )[0]
-        return self.doc_details(results, include_text)
+        sources = self.doc_details(results, include_text)
+        return self.retrieve_unique_pages(sources)
+
+    def retrieve_unique_pages(self, sources):
+        """
+        Get 3 unique pages from the sources retrieved from the retriever
+        """
+        unique_sources = []
+        top_pages = []
+        for source in sources:
+            if source['source'] not in unique_sources:
+                unique_sources.append(source)
+            if len(unique_sources) == 3:
```

Comment:
I think we can use the `RAG_MAX_DOCUMENTS` setting here:
```suggestion
            if len(unique_sources) == settings.RAG_MAX_DOCUMENTS,:
```